### PR TITLE
SL-1111 delete hearing part endpoint

### PR DESCRIPTION
**Pull request by filipczyks**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SL-1111

### Change description ###
added delete hearing part endpoint to events

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/HearingPartControllerTest.java [hunk 1]
```diff
@@ -105,6 +105,19 @@ public void upsertHearingPart_savesHearingPartToService() throws Exception {
...
+    @Test
+    public void deleteHearingPart_deletesHearingPart() throws Exception {
+        val id = UUID.randomUUID();
+
+        val serviceResult = createHearingPart();
+        serviceResult.setDeleted(true);
+        when(hearingPartService.deleteHearingPart(id)).thenReturn(serviceResult);
+
+        val response = mvc.deleteAndReturnResponse(URL + "/" + id);
+
```
Comment (satyachundur) [inline src/test/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/HearingPartControllerTest.java]: can we not check response code as well for this deletion.?  i don't know whether we do same for other controller test.

Comment (filipczyks) [reply src/test/java/uk/gov/hmcts/reform/sandl/snlevents/controllers/HearingPartControllerTest.java]: status code is checked in mvc.deleteAndReturnResponse(URL + "/" + id)

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingPartService.java [hunk 2]
```diff
@@ -131,4 +132,11 @@ private UserTransactionData getLockedSessionTransactionData(UUID id) {
                 "unlock",
                 0);
     }
+
+    public HearingPart deleteHearingPart(UUID hearingPartId) {
+        val hearingPart = hearingPartRepository.findOne(hearingPartId);
```
[the remark this answers is not in the document]
Comment (filipczyks) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingPartService.java]: not valid anymore, as I need to fetch HearingPart to send it to snl-rules

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingPartService.java [hunk 3]
```diff
@@ -131,4 +132,11 @@ private UserTransactionData getLockedSessionTransactionData(UUID id) {
...
+        hearingPart.setDeleted(true);
+
```
[the remark this answers is not in the document]
Comment (filipczyks) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingPartService.java]: thanks, done

[the remark this answers is not in the document]
Comment (r-kruk-kainos) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/HearingPartService.java]: You still have to change PostDbDataController

**Comment by r-kruk-kainos** [comment]
You still have to change PostDbDataController

**Pull request merged**